### PR TITLE
Added acal.rb, acal_view.rb. Modified authentication.rb and config.yaml ...

### PR DESCRIPTION
**Pull request by kgirvan**
Added acal.rb, acal_view.rb. Modified authentication.rb and config.yaml for acal. Added config/data/ks/common/text directory.

### CODE REVIEW

File: lib/ks/acal/acal.rb [hunk 1]
```diff
@@ -0,0 +1,395 @@
...
+
+    puts "context = #{@request.config.context}"
+    
+    # click "Enrollment Home"
+    @request.add("/?channelTitle=Enrollment%20Home&channelUrl=#{@request.url}/#{opts[:kr_name]}/launch?viewId=enrollmentHomeView&methodToCall=start")
+    @request.add("/#{opts[:kr_name]}/launch?viewId=enrollmentHomeView&methodToCall=start")  
+
+    @request.add_thinktime(10)
+
+    @search_form_key = 'search_form_key'
```
Comment (kcampos) [inline lib/ks/acal/acal.rb]: Why is this a class variable?

File: lib/ks/acal/acal.rb [hunk 2]
```diff
@@ -0,0 +1,395 @@
...
+    @request.add_thinktime(10)
+
+    opts[:file_path] = "#{opts[:data_dir]}/#{opts[:directory]}/#{opts[:type]}/"
+
+    @searchFileName= "searchPost.txt"
+    createContentsFile(opts[:file_path], @searchFileName, contents_search)
+    @searchPathAndFileName = opts[:file_path] +  @searchFileName
+
+    @view_form_key = "view_form_key"
+
```
Comment (kcampos) [inline lib/ks/acal/acal.rb]: again I'm not understanding why these are class variables

File: lib/ks/acal/acal.rb [hunk 3]
```diff
@@ -0,0 +1,395 @@
...
+%%_#{opts[:view_focusId_dyn_var_name]}%%
+#{bodyBoundaryView}
+Content-Disposition: form-data; name=\"jumpToId\"
+
+%%_#{opts[:view_jumpToId_dyn_var_name]}%%
+#{bodyBoundaryView}--
+"
+    # the following is necessary to ensure that a CRLF happens at the end of each line
+    contents_view=contents_view.gsub("\n","\r\n")
```
Comment (kcampos) [inline lib/ks/acal/acal.rb]: simpler way of expressing this is: contents_view.gsub!("\n","\r\n")

File: lib/ks/acal/acal.rb [hunk 4]
```diff
@@ -0,0 +1,395 @@
...
+    
+  # Main Menu homepage
+  def homepage
+            
+  end
+  
+  
+  # View an Academic Calendar
+  # TODO:
+  #
```
Comment (kcampos) [inline lib/ks/acal/acal.rb:41]: Need to add a lot of comments here, this is a big confusing method. Please follow the rdoc doc pattern and at least address the params that you expect callers may specify.

File: lib/ks/acal/acal.rb (hunk 1 above)
Comment (kgirvan) [inline lib/ks/acal/acal.rb]: It should not be a class variable.  I originally had 3 methods one for clicking "Enrollment Home" button, one for search and one for view, then I put everything into the acal_view method.

File: lib/ks/acal/acal.rb [hunk 5]
```diff
@@ -0,0 +1,410 @@
...
+    @request.add_thinktime(10)
+
+  end
+
+
+  private
+
+
+  def createContentsFile(filepath,filename,contents_str)
+  # ==synopsis
```
Comment (kcampos) [inline lib/ks/acal/acal.rb]: Why is this inside the method declaration and not above?

Comment (kgirvan) [inline lib/ks/acal/acal.rb]: Moved comments above method

**Pull request merged**